telegram/ext/_updater.py

```diff
@@ -281,170 +330,200 @@ def start_webhook(
+    async def _network_loop_retry(
+        self,
+        action_cb: Callable[..., Coroutine],
+        on_err_cb: Callable[[TelegramError], None],
+        description: str,
+        interval: float,
+    ) -> None:
         """Perform a loop calling `action_cb`, retrying after network errors.
 
         Stop condition for loop: `self.running` evaluates :obj:`False` or return value of
         `action_cb` evaluates :obj:`False`.
 
         Args:
             action_cb (:obj:`callable`): Network oriented callback function to call.
-            onerr_cb (:obj:`callable`): Callback to call when TelegramError is caught. Receives the
-                exception object as a parameter.
+            on_err_cb (:obj:`callable`): Callback to call when TelegramError is caught. Receives
+                the exception object as a parameter.
             description (:obj:`str`): Description text to use for logs and exception raised.
             interval (:obj:`float` | :obj:`int`): Interval to sleep between each call to
                 `action_cb`.
 
         """
-        self.logger.debug('Start network loop retry %s', description)
+        self._logger.debug('Start network loop retry %s', description)
         cur_interval = interval
         while self.running:
             try:
-                if not action_cb():
-                    break
-            except RetryAfter as exc:
-                self.logger.info('%s', exc)
-                cur_interval = 0.5 + exc.retry_after
-            except TimedOut as toe:
-                self.logger.debug('Timed out %s: %s', description, toe)
-                # If failure is due to timeout, we should retry asap.
-                cur_interval = 0
-            except InvalidToken as pex:
-                self.logger.error('Invalid token; aborting')
-                raise pex
-            except TelegramError as telegram_exc:
-                self.logger.error('Error while %s: %s', description, telegram_exc)
-                onerr_cb(telegram_exc)
-                cur_interval = self._increase_poll_interval(cur_interval)
-            else:
-                cur_interval = interval
+                try:
+                    if not await action_cb():
+                        break
+                except RetryAfter as exc:
+                    self._logger.info('%s', exc)
+                    cur_interval = 0.5 + exc.retry_after
+                except TimedOut as toe:
+                    self._logger.debug('Timed out %s: %s', description, toe)
+                    # If failure is due to timeout, we should retry asap.
+                    cur_interval = 0
+                except InvalidToken as pex:
+                    self._logger.error('Invalid token; aborting')
+                    raise pex
+                except TelegramError as telegram_exc:
+                    self._logger.error('Error while %s: %s', description, telegram_exc)
+                    on_err_cb(telegram_exc)
```

Comment:
I didn't allow for coroutines here because awaiting them would block the network retry loop. that's basically also why previously we had this thing, where we put the errors into the `update_queue` such that dispatcher would handle them in it's own thread as to not block the loop.
so I'd rather clarify that the `error_callback` must not be a coroutine function.

Reply:
Maybe raise `ValueError` if it is a coroutine just like how did for CBQH?